### PR TITLE
chore: Remove config & library option merging

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This change removes config & library option merging from v6. 

#### Description of how you validated changes
Tested in a sample app:
```
Amplify.configure(config, { ssr: true });

// Configure Analytics providers which are not managed by the CLI
Amplify.configure({
    ...Amplify.getConfig(),
    Analytics: {
        Kinesis: {
            region: 'us-west-2'
        },
        KinesisFirehose: {
            region: 'us-west-2'
        },
        Personalize: {
            trackingId: 'f1b2d240-f7e7-416a-af88-759d7e258994',
            region: 'us-west-2'
        }
    }
})
```

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/singleton/Amplify.ts [hunk 1]
```diff
@@ -44,15 +44,8 @@ export class AmplifyClass {
...
-			this.resourcesConfig,
-			resolvedResourceConfig
-		);
-
-		this.libraryOptions = mergeLibraryOptions(
-			this.libraryOptions,
-			libraryOptions
-		);
+		this.resourcesConfig = resolvedResourceConfig;
+		this.libraryOptions = libraryOptions;
```
Comment (AllanZhengYP) [inline packages/core/src/singleton/Amplify.ts:48]: Can we NOT assign it if it's undefined? So the `ssr: true` won't be flipped in following example:
```js
Amplify.configure(config, { ssr: true });

// Configure Analytics providers which are not managed by the CLI
Amplify.configure({
    ...Amplify.getConfig(),
    Analytics: {
        Kinesis: {
            region: 'us-west-2'
        },
        KinesisFirehose: {
            region: 'us-west-2'
        },
        Personalize: {
            trackingId: 'f1b2d240-f7e7-416a-af88-759d7e258994',
            region: 'us-west-2'
        }
    }
})
```

Comment (jimblanc) [reply packages/core/src/singleton/Amplify.ts:48]: Ack good catch

**Review by stocaaro — DISMISSED** [review]

**Review by AllanZhengYP — DISMISSED** [review]
🚀🚀🚀

**Review by stocaaro — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

File: packages/core/src/singleton/Amplify.ts [hunk 2]
```diff
@@ -44,15 +48,11 @@ export class AmplifyClass {
...
-		);
+		this.resourcesConfig = resolvedResourceConfig;
 
-		this.libraryOptions = mergeLibraryOptions(
-			this.libraryOptions,
-			libraryOptions
-		);
+		if (libraryOptions) {
+			this.libraryOptions = libraryOptions;
+		}
```
Comment (elorzafe) [inline packages/core/src/singleton/Amplify.ts:55]: If this doesnt happen it will not replace `libraryOptions` is that ok?

Comment (jimblanc) [reply packages/core/src/singleton/Amplify.ts:55]: Yep it should be, `libraryOptions` defaults to `{}` when class is initialized

Comment (jimblanc) [reply packages/core/src/singleton/Amplify.ts:55]: Will make triple sure that the default library options are present in the sample app, but I was able to login & do auth things so I assume the token & cred providers were there

Comment (HuiSF) [reply packages/core/src/singleton/Amplify.ts:55]: If I called `Amplify.configure()` with `libraryOptions`.

Then I call it again to reconfigure, and want to remove the previously configured `libraryOptions`, this check will prevent me from doing so correct (explicitly setting as `undefined` in the second call)?

Comment (jimblanc) [reply packages/core/src/singleton/Amplify.ts:55]: Yeah it's good

**Review by elorzafe — APPROVED** [review]

**Pull request merged**